src/main/java/casper/components/Processor.java

```diff
@@ -0,0 +1,122 @@
+package casper.components;
+
+import casper.CasperChat;
+import casper.exceptions.ExceptionWithSolution;
+import casper.exceptions.CustomIOException;
+import casper.exceptions.MissingArgumentException;
+import casper.exceptions.InvalidCommandException;
+import casper.tasks.Deadline;
+import casper.tasks.Event;
+import casper.tasks.ToDo;
+
+import java.util.Objects;
+
+import static casper.utils.DateTimeUtils.convertStringToDateTime;
+
+/**
+ * Processes user input commands and manages tasks in the task list.
+ */
+public class Processor {
+
+    /**
+     * Processes the user input command and updates the task list accordingly.
+     *
+     * @param input The user input command to process.
+     * @param taskList The task list to which the command will be applied.
+     * @throws ExceptionWithSolution If there is an issue with the command that needs a solution.
+     * @throws CustomIOException If an I/O error occurs while interacting with the storage.
+     * @throws MissingArgumentException If required arguments are missing for a specific command.
+     * @throws InvalidCommandException If the command is invalid or not recognized.
+     */
+    public static void processInput(String input, TaskList taskList) throws
+            ExceptionWithSolution, CustomIOException {
+
+        String[] inputParts = Parser.parseInputToCommandAndArgument(input);
+        String command = inputParts[0];
+        String argument = inputParts.length > 1 ? inputParts[1] : null;
+
+        switch (command) {
+            case "list" -> {
```

Comment:
Should case be on the same line as switch based on the coding standard?

Reply:
I'll make the necessary changes. Thanks!